pocs/utils/rs232.py

```diff
@@ -12,56 +12,66 @@
 
 
 class SerialData(PanBase):
-
     """
     Main serial class
     """
 
-    def __init__(self, port=None, baudrate=115200, threaded=True, name="serial_data"):
+    def __init__(self,
+                 port=None,
+                 baudrate=115200,
+                 threaded=True,
+                 name="serial_data",
+                 open_delay=2.0):
         PanBase.__init__(self)
 
-        try:
-            self.ser = serial.Serial()
-            self.ser.port = port
-            self.ser.baudrate = baudrate
-            self.is_threaded = threaded
-
-            self.ser.bytesize = serial.EIGHTBITS
-            self.ser.parity = serial.PARITY_NONE
-            self.ser.stopbits = serial.STOPBITS_ONE
-            self.ser.timeout = 1.0
-            self.ser.xonxoff = False
-            self.ser.rtscts = False
-            self.ser.dsrdtr = False
-            self.ser.write_timeout = False
-            self.ser.open()
-
-            self.name = name
-            self.queue = deque([], 1)
-            self._is_listening = False
-            self.loop_delay = 2.
-
-            if self.is_threaded:
-                self._serial_io = TextIOWrapper(BufferedRWPair(self.ser, self.ser),
-                                                newline='\r\n', encoding='ascii', line_buffering=True)
+        self.ser = serial.serial_for_url(port, do_not_open=True)
```

Comment:
Not sure what "this" refers to. If you mean do_not_open, the advantage is that you set all of the properties (e.g. baud rate and byte size) before opening the port. Depending on the device, it may require effectively closing and reopening to change properties (_reconfigure_port is called for this).

Reply:
Sorry, bad comment on my part. This was in reference to using `serial_for_url`. I reviewed this out of order, starting with this file, so after looking at how the tests are set up this makes more sense.